### PR TITLE
Validate user email - redone, change package names to singular forms

### PR DESCRIPTION
**Pull request by gyokhan28**

### CODE REVIEW

**Review by alex-ganchev — APPROVED** [review]
The code looks fine to me, and functionally everything works.

File: Deploy/.env.example [hunk 1]
```diff
@@ -0,0 +1,7 @@
+JWT_SECRET_KEY=
```
Comment (emiliyank) [inline Deploy/.env.example:1]: Very good that you add such file.

File: Deploy/docker-compose.yml [hunk 2]
```diff
@@ -7,9 +7,9 @@ services:
     ports:
       - '5432:5432'
     environment:
-      POSTGRES_USER: 'postgres'
-      POSTGRES_PASSWORD: '**your-postgres-pass-here**'
-      POSTGRES_DB: 'b2b-opportunities'
+      POSTGRES_USER: ${DB_USERNAME}
```
Comment (emiliyank) [inline Deploy/docker-compose.yml:10]: It's good that we use environment variables here.

File: src/main/java/com/example/b2b_opportunities/Service/MailService.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+public class MailService {
+
+    private final ConfirmationTokenService confirmationTokenService;
+    private final JavaMailSender mailSender;
+    @Value("${spring.mail.username}")
+    private String fromMail;
+
+    private String generateConfirmationLink(User user) {
+        String token = confirmationTokenService.generateConfirmationCode(user);
+        return "<a href=http://localhost:8082/api/auth/register/confirm?token=" + token + ">Confirm Email</a>";
```
Comment (Valyo96) [inline src/main/java/com/example/b2b_opportunities/Service/MailService.java:24]: public String generateConfirmationLink(User user, HttpServletRequest request) {
    String token = confirmationTokenService.generateConfirmationCode(user);
    
    // Extract the dynamic base URL from the request
    String baseUrl = request.getScheme() + "://" + request.getServerName() + ":" + request.getServerPort() + request.getContextPath();;
    
    return "<a href=" + baseUrl + "/api/auth/register/confirm?token=" + token + ">Confirm Email</a>";
} 
You have to make sure you get the url dynamically.

**Review by emiliyank — APPROVED** [review]

File: src/main/java/com/example/b2b_opportunities/Service/MailService.java (hunk 3 above)
Comment (gyokhan28) [reply src/main/java/com/example/b2b_opportunities/Service/MailService.java]: Thanks, the URL is now generated dynamically.

**Pull request merged**